src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,42 @@
+package mate.academy.service.impl;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.AuthenticationService;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> userOptional = userService.findByEmail(email);
+        if (userOptional.isEmpty()) {
+            throw new AuthenticationException("Login or password is incorrect");
+        }
+        User user = userOptional.get();
+        String hashPassword = HashUtil.hashPassword(password, user.getSalt());
+        if (user.getPassword().equals(hashPassword)) {
+            return user;
+        }
+        throw new AuthenticationException("Login or password is incorrect");
+    }
+
+    @Override
+    public User register(String email, String password) throws RegistrationException {
+        if (userService.findByEmail(email).isPresent()) {
+            throw new RegistrationException("Account with email: " + email + " already exist");
+        }
+        if (email.isEmpty() || password.isEmpty()) {
+            throw new RegistrationException("Email and password can`t be empty");
+        }
+        return userService.add(new User(email, password));
+    }
```

Comment:
The register method should go first. It is more consistent. Also, check other places where it can occur